src/OpenLoco/Orders.h

```diff
@@ -0,0 +1,171 @@
+#pragma once
+#include "Map/Tile.h"
+#include "Types.hpp"
+#include <iterator>
+
+namespace OpenLoco::Map
+{
+    struct map_pos3;
+}
+namespace OpenLoco::Vehicle
+{
+
+#pragma pack(push, 1)
+    enum class OrderType
+    {
+        End,
+        StopAt,
+        RouteThrough,
+        RouteWaypoint,
+        UnloadAll,
+        WaitFor
+    };
+
+    struct Order
+    {
+        uint8_t _type;
+        OrderType getType() const { return OrderType(_type & 0x7); }
+        uint32_t getOffset() const;
+        template<typename T>
+        constexpr bool is() const { return getType() == T::TYPE; }
+        template<typename T>
+        T* as() { return is<T>() ? reinterpret_cast<T*>(this) : nullptr; }
+        template<typename T>
+        const T* as() const { return is<T>() ? reinterpret_cast<const T*>(this) : nullptr; }
+    };
+
+    static_assert(sizeof(Order) == 1, "Size of order must be 1 for pointer arithmatic to work in OrderTableView");
+
+    struct OrderEnd : Order
+    {
+        static constexpr OrderType TYPE = OrderType::End;
+    };
+
+    struct OrderStation : Order
+    {
+        uint8_t _1;
+
+        station_id_t getStation() const
+        {
+            return ((_type & 0xC) << 2) | _1;
+        }
+        void setFormatArguments(FormatArguments& args) const;
+    };
+
+    struct OrderStopAt : OrderStation
+    {
+        static constexpr OrderType TYPE = OrderType::StopAt;
+    };
+
+    struct OrderRouteThrough : OrderStation
+    {
+        static constexpr OrderType TYPE = OrderType::RouteThrough;
+    };
+
+    struct OrderRouteWaypoint : Order
+    {
+        static constexpr OrderType TYPE = OrderType::RouteWaypoint;
+        uint8_t _1;
+        uint8_t _2;
+        uint8_t _3;
+        uint8_t _4;
+        uint8_t _5;
```

Comment:
Does the naming vary by waypoint? I'm assuming these are placeholders. If not, think we should name these `var_0X` at least.

Reply:
I was copying the style from the tile elements. None of these fields should be accessed directly.

Reply:
Right, but surely they have some semantics? Like X/Y/Z tile coordinates and such?